### ISSUE TITLE
`bad-dunder-method-name` (`PLW3201`) - false positive on `__replace__` dunder

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#14145]
```py
from copy import replace


class Foo:
    def __replace__(self, *args: object, **kwargs: object): # PLW3201: Dunder method `__replace__` has no special meaning in Python 3
        print("hi")


replace(Foo()) # calls `Foo.__replace__`
```
[playground](https://play.ruff.rs/e6bae1cb-f4c9-4929-97e3-cc74a45eb0c2)
